diagnostic.js

```diff
@@ -32,25 +32,89 @@ const Movement = require('./models/movement.js');
 
 /// ADD YOUR CODE BELOW
 
-const create = (name, description, startYear, endYear) => {};
+// in the lesson we used db.close() - are we not doing this here because we
+// don't want to close our server?
+
+const create = (name, description, startYear, endYear) => {
+  Movement.create({
+    name: name,
+    description: description,
+    startYear: startYear,
+    endYear: endYear
+  }).then(function(movement){
+    console.log(movement);
+  }).catch(function(error){
+    console.error(error);
+  });
+};
 // Success -> console.log new Movement as JSON
 // Failure -> Console.error
 
-const index = () => {};
+const index = () => {
+  let search = {};
+  if (arguments[0] && arguments[1]) {
+  let field = arguments[0];
+  let criterion = arguments[0];
+  if (criterion[0] === '/') {
+    let regex = new RegExp(criterion.slice(1, criterion.length -1));
+    search[field] = regex;
+  } else {
+    search[field] = criterion;
+  }
+}
+
+  Movement.find(search).then(function(movement){
```

Comment:
`movement` is used as name of the parameter in this callback function and also in the next line.  Try to avoid doing this in the future.  `movements` is a better name since the return value of `find()` is a collection.